app/src/main/java/io/gnosis/safe/ui/transaction/TransactionsViewModel.kt

```diff
@@ -15,29 +19,46 @@ class TransactionsViewModel
 ) : BaseStateViewModel<TransactionsViewState>(appDispatchers) {
 
     override fun initialState(): TransactionsViewState = TransactionsViewState(null, true)
-
     fun load() {
         safeLaunch {
             val safeAddress = safeRepository.getActiveSafe()!!.address
             val transactions = transactionRepository.getTransactions(safeAddress)
             updateState {
+                val newTransactions = newTransactions(transactions, safeAddress)
+                val transactionsWithSectionHeaders = addSectionHeaders(newTransactions)
                 TransactionsViewState(
                     isLoading = false,
-                    viewAction = LoadTransactions(transactions.results.mapNotNull { transaction ->
-                        when (transaction) {
-                            is Transaction.Transfer -> TransactionView.Transfer(transaction, transaction.recipient == safeAddress)
-                            is Transaction.SettingsChange -> TransactionView.SettingsChange(transaction)
-                            else -> null
-                        }
-                    })
+                    viewAction = LoadTransactions(transactionsWithSectionHeaders)
                 )
             }
         }
     }
-}
 
-sealed class TransactionView(open val transaction: Transaction) {
+    private fun newTransactions(
+        transactions: Page<Transaction>,
+        safeAddress: Solidity.Address
+    ): List<TransactionView> {
+        return transactions.results.mapNotNull { transaction ->
+            when (transaction) {
+                is Transaction.Transfer -> TransactionView.Transfer(transaction, transaction.recipient == safeAddress)
+                is Transaction.SettingsChange -> TransactionView.SettingsChange(transaction)
+                else -> null
+            }
+        }
+    }
+
+    private fun addSectionHeaders(newTransactions: List<TransactionView>): List<TransactionView> {
+        val mutableList = newTransactions.toMutableList()
+        //TODO: Find first QUEUED tx and insert SectionHeader before it
+        mutableList.add(0, TransactionView.SectionHeader(title = R.string.tx_list_queue))
+        //TODO: Find first HISTORY tx and insert SectionHeader before it
+        mutableList.add(3, TransactionView.SectionHeader(title = R.string.tx_list_history))
```

Comment:
we can't assume it will be at position 3 all the time

Reply:
Of course. That is what the TODO is for. As soon as we have the status field in Transactions, we can find the right spot.

Maybe I'll just wait for Joses PR? And then amend this?

Reply:
yes, let's review & merge Joses PR first. It is already open